docs/api.rst

```diff
@@ -0,0 +1,266 @@
+.. _ksql-http-api:
+
+KSQL HTTP API Reference
```

Comment:
@rodesai : Can we please call this the KSQL REST API, not the KSQL HTTP API?  We have talked about this already in a different context.

Reply:
I tried to avoid calling it that since its not really a REST API. Are we sure that's what we want to call it?